src/main/java/christmas/domain/EventType.java

```diff
@@ -0,0 +1,60 @@
+package christmas.domain;
+
+import christmas.service.event.EventDayCalculator;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+public enum EventType {
+    CHRISTMAS("크리스마스 디데이 할인", EventDayCalculator.christmasDay(), orderDay -> {
+        final int basicDiscountAmount = 1000;
+        final int discountAmountUnit = 100;
+        return (basicDiscountAmount + discountAmountUnit * (orderDay - 1)) * -1;
+    }),
+    WEEKDAY("평일 할인", EventDayCalculator.weekDay(), dessertCount -> {
+        final int discountAmountUnit = 2023;
+        return (discountAmountUnit * dessertCount) * -1;
+    }),
+    WEEKEND("주말 할인", EventDayCalculator.weekEnd(), mainCourseCount -> {
+        final int discountAmountUnit = 2023;
+        return (discountAmountUnit * mainCourseCount) * -1;
+    }),
+    SPECIAL("특별 할인", EventDayCalculator.specialDay(), anything -> {
+        return -1000;
+    });
+
+    private final String eventName;
+    private final List<Integer> possibleDays;
+    private final Function<Integer, Integer> calculator;
+
+    EventType(String eventName, List<Integer> possibleDays, Function<Integer, Integer> calculator) {
+        this.eventName = eventName;
+        this.possibleDays = possibleDays;
+        this.calculator = calculator;
+    }
+
+    public static List<EventType> getApplicableEventTypes(int targetDay) {
+        List<EventType> applicableEventTypes = new ArrayList<>();
+        for (EventType eventType : values()) {
+            if (eventType.possibleDays.contains(targetDay)) {
+                applicableEventTypes.add(eventType);
+            }
+        }
+        return applicableEventTypes;
```

Comment:
이 부분을 Stream으로 처리하고 toList() 메서드를 사용하면 훨씬 간결해 지면서 가독성도 좋아질 것이라 생각해요!
추가적으로 혜택 적용이 가능한지 확인하는 부분도 메서드로 추출하면 더 좋을 것 같고요!
```suggestion
    public static List<EventType> getApplicableEventTypes(int targetDay) {
        return Arrays.stream(EventType.values())
                .filter(eventType -> eventType.appliable(targetDay))
                .toList();
    }

    public boolean appliable(int targetDay){
        return possibleDays.contains(targetDay);
    }
```

Reply:
훨씬 가독성이 좋은 코드라 생각됩니다!! 좋은 팁 감사합니다 :)